### ISSUE TITLE
[FR] Improve logging for timeout error

### ISSUE THREAD

**Issue by chemelli74** [StyraHem/ShellyForHASS#308]
## Is your feature request related to a problem? Please describe.

Timeout spam log and miss device information making hard to identify offending device.

## Describe the solution you'd like

One single log line with device info, something like

`2020-05-24 23:38:03 ERROR (CoAP) [pyShelly] Error http GET: http://192.168.188.55/settings, timed out for device 'Shelly Livingroom Light'`

## Describe alternatives you've considered

None

## Additional context

```
2020-05-24 23:38:03 ERROR (CoAP) [pyShelly] Error http GET: http://192.168.188.55/settings, timed out, Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/pyShelly/utils.py", line 57, in shelly_http_get
    resp = conn.getresponse()
  File "/usr/local/lib/python3.7/http/client.py", line 1344, in getresponse
    response.begin()
  File "/usr/local/lib/python3.7/http/client.py", line 306, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.7/http/client.py", line 267, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/local/lib/python3.7/socket.py", line 589, in readinto
    return self._sock.recv_into(b)
socket.timeout: timed out
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/pyShelly/utils.py", line 57, in shelly_http_get
    resp = conn.getresponse()
  File "/usr/local/lib/python3.7/http/client.py", line 1344, in getresponse
    response.begin()
  File "/usr/local/lib/python3.7/http/client.py", line 306, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.7/http/client.py", line 267, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/local/lib/python3.7/socket.py", line 589, in readinto
    return self._sock.recv_into(b)
socket.timeout: timed out
```

**Comment by pdonat01** [comment]
same error here,
But i have this:
http GET: http://192.168.0.21/settings
but this ip is a router ip not a home assistant ip. Where can i change? i tried to reinstall shelly integration but error is same

**Comment by hakana** [comment]
> same error here,
> But i have this:
> http GET: http://192.168.0.21/settings
> but this and old router ip not the home assistant ip, or new router ip. Where can i change? i tried to reinstall shelly integration but error is same

Is the ip the ip of your Shelly?

**Comment by pdonat01** [comment]
no, i have 2 router in my network. Router A ip: 192.168.0.21 router B ip: 192.168.0.1
home assistant is connected to B network via 192.168.0.13 ip. I dont understand why try to connect to via 192.168.0.21. It has nothing to do with the other network.

**Comment by hakana** [comment]
What is the protocols attribute on your device?

**Comment by pdonat01** [comment]
Where can i find this attribute?

**Comment by hakana** [comment]
![image](https://user-images.githubusercontent.com/1168654/84154064-e692fc00-aa66-11ea-9b75-7c8204b30e98.png)

**Comment by pdonat01** [comment]
Unfortunately my error occurs when home assitant is started and try to load the integration. So i havent entites, where i can check this info.

**Comment by denniswham** [comment]
I see the exact same error in my log for a cover. I have three covers, two working fine, one is showing unavailable on and off since yesterday night.

`Logger: pyShelly
Source: /usr/local/lib/python3.7/site-packages/pyShelly/utils.py:37
First occurred: 8:53:16 PM (1 occurrences)
Last logged: 8:53:16 PM

Error http GET: http://192.168.178.112/settings, timed out, Traceback (most recent call last): File "/usr/local/lib/python3.7/site-packages/pyShelly/utils.py", line 57, in shelly_http_get resp = conn.getresponse() File "/usr/local/lib/python3.7/http/client.py", line 1344, in getresponse response.begin() File "/usr/local/lib/python3.7/http/client.py", line 306, in begin version, status, reason = self._read_status() File "/usr/local/lib/python3.7/http/client.py", line 267, in _read_status line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1") File "/usr/local/lib/python3.7/socket.py", line 589, in readinto return self._sock.recv_into(b) socket.timeout: timed out
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/pyShelly/utils.py", line 57, in shelly_http_get
    resp = conn.getresponse()
  File "/usr/local/lib/python3.7/http/client.py", line 1344, in getresponse
    response.begin()
  File "/usr/local/lib/python3.7/http/client.py", line 306, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.7/http/client.py", line 267, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/usr/local/lib/python3.7/socket.py", line 589, in readinto
    return self._sock.recv_into(b)
socket.timeout: timed out`

192.168.178.112 is the IP address of the Shelly device.

**Comment by pdonat01** [comment]
I think i found a bug. If raspberry is in another network(A), and shelly is in another network(B). the integration try to connect the shelly device between 2 router but there is no found a shelly device just default gateway address.
Because my error log contains this ip: http://192.168.0.21/settings and this ip is my B router address.

**Comment by denniswham** [comment]
That's not the case for me. My RPi and Shelly devices are in the same network and they can see each other, and the IP address in my log is the Shelly device's IP. Sometimes after a HA reboot the device is available again, and sometimes it disappears again.

**Comment by denniswham** [comment]
Since I only have this issue with one device, I have factory reset it today, to no avail.

**Comment by denniswham** [comment]
I'm quite sure it's not related to the integration. If I navigate to the web interface of the Shelly, it takes a really long time after I put the credentials before I see the UI. Longer than with any of the other Shelly's. Once I see the UI and restart the HA, the entity is available in HA as well. SO it's just a matter of HA not having access to the Shelly because of the Shelly itself, hence the timeout. I believe this issue can be safely closed.

**Comment by denniswham** [comment]
See today's announcement in the Shelly support group (English Version) Facebook support group, subject "Device WI-Fi disconnect - BETA program." I have updated my affected Shelly 2.5, see if anything will be improved over the next couple of days.

**Comment by chemelli74** [comment]
Closed via PR [#26 on pyShelly](https://github.com/StyraHem/pyShelly/pull/26)

Simone
